patches/kexec-2.0.12.patch

```diff
@@ -0,0 +1,67 @@
+diff -u --recursive clean/kexec-tools-2.0.12/kexec/arch/i386/kexec-x86-common.c build/kexec-tools-2.0.12/kexec/arch/i386/kexec-x86-common.c
+--- clean/kexec-tools-2.0.12/kexec/arch/i386/kexec-x86-common.c	2015-11-29 18:52:07.000000000 -0500
++++ kexec-tools-2.0.12/kexec/arch/i386/kexec-x86-common.c	2017-01-04 13:46:53.684313636 -0500
+@@ -24,6 +24,7 @@
+ #include <fcntl.h>
+ #include <stddef.h>
+ #include <stdio.h>
++#include <inttypes.h>
+ #include <errno.h>
+ #include <stdint.h>
+ #include <string.h>
+@@ -74,20 +75,20 @@
+ 		return -1;
+ 	}
+ 	while(fgets(line, sizeof(line), fp) != 0) {
+-		unsigned long long start, end;
++		uintptr_t start, end;
+ 		char *str;
+ 		int type;
+ 		int consumed;
+ 		int count;
+ 		if (memory_ranges >= MAX_MEMORY_RANGES)
+ 			break;
+-		count = sscanf(line, "%Lx-%Lx : %n",
++		count = sscanf(line, "%"SCNxPTR"-%"SCNxPTR" : %n",
```

Comment:
nope, it's not a musl bug, it's a conformance bug in GLIBC and a bug for apps that use it.
according to POSIX (man 3p fscanf)
```
     L    Specifies  that  a  following  a,  A, e, E, f, F, g, or G conversion
          specifier applies to an argument with type pointer to long double.
```
the author wanted to actually use "%llx" but he did it the GLIBC-only-crash-everywhere-else way.
i've fixed this in various linux-specific opensource projects, the most prominent being [alsa-lib](http://git.alsa-project.org/?p=alsa-lib.git;a=commitdiff;h=1d3f7975f920f47e6a8a324f547da2180e64171a)